### PR TITLE
Replace dummy sw.js with pre-caching and runtime-caching workbox SW

### PR DESCRIPTION
**Pull request by t3chguy**
Service Worker

Fixes https://github.com/vector-im/element-web/issues/12591

### CODE REVIEW

**Comment by turt2live** [comment]
> Service Worker

So many words.

File: src/vector/index.html [hunk 1]
```diff
@@ -55,16 +55,6 @@
...
-    <img src="<%= require('matrix-react-sdk/res/img/feather-customised/warning-triangle.svg') %>" width="24" height="23" style="visibility: hidden; position: absolute; top: 0px; left: 0px;"/>
-    <img src="<%= require('matrix-react-sdk/res/img/format/bold.svg') %>" width="25" height="22" style="visibility: hidden; position: absolute; top: 0px; left: 0px;"/>
-    <img src="<%= require('matrix-react-sdk/res/img/format/code.svg') %>" width="25" height="22" style="visibility: hidden; position: absolute; top: 0px; left: 0px;"/>
-    <img src="<%= require('matrix-react-sdk/res/img/format/italics.svg') %>" width="25" height="22" style="visibility: hidden; position: absolute; top: 0px; left: 0px;"/>
-    <img src="<%= require('matrix-react-sdk/res/img/format/quote.svg') %>" width="25" height="22" style="visibility: hidden; position: absolute; top: 0px; left: 0px;"/>
-    <img src="<%= require('matrix-react-sdk/res/img/format/strikethrough.svg') %>" width="25" height="22" style="visibility: hidden; position: absolute; top: 0px; left: 0px;"/>
-
```
Comment (turt2live) [inline src/vector/index.html:67]: most of these are required so the user actually sees an icon when they go offline, such as the warning triangle. The composer ones are so there isn't a delay when showing the formatting bar the first time

**Review by turt2live — CHANGES_REQUESTED** [review]
other than the prefetch, this lgtm

File: src/vector/index.html (hunk 1 above)
Comment (t3chguy) [reply src/vector/index.html:67]: Yup, now handled by the service worker's prefetch

Comment (turt2live) [reply src/vector/index.html:67]: was it committed? I'm not seeing it

Comment (t3chguy) [reply src/vector/index.html:67]: It is the behaviour of workbox to precache all webpack outputs which match criteria like the above

Comment (turt2live) [reply src/vector/index.html:67]: ah, neat. Would be good to include that in the PR description :)

**Review by turt2live — APPROVED** [review]

**Comment by t3chguy** [comment]
Blocking until after RC as it is risky

**Pull request merged**